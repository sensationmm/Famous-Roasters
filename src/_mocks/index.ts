import { CatalogueMockError, CatalogueMockMissingData, CatalogueMocks, CatalogueMocksPagination } from './CatalogueMock'
import { ProductMock } from './ProductMock'

export { CatalogueMocks as CatalogueMocks }
export { CatalogueMocksPagination as CatalogueMocksPagination }
export { CatalogueMockError as CatalogueMockError }
export { CatalogueMockMissingData as CatalogueMockMissingData }
export { ProductMock as ProductMock }
