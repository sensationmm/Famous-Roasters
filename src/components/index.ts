import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
