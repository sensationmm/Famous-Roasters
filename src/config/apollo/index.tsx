import { ApolloClient, ApolloLink, createHttpLink, from, InMemoryCache, NormalizedCacheObject } from '@apollo/client'
import { onError } from '@apollo/client/link/error'

export const apolloClient = (): ApolloClient<NormalizedCacheObject> => {
  const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors)
      graphQLErrors.forEach(({ message: errorMessage, locations, path }) => {
        console.log(`[GraphQL error]: Message: ${errorMessage}, Location: ${locations}, Path: ${path}`)
      })

    if (networkError) console.log(`[Network error]: ${networkError}`)
  })

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: from([
      errorLink,
      createHttpLink({
        uri: process.env.REACT_APP_SHOPIFY_STOREFRONT_GRAPHQL_ENDPOINT,
        headers: {
          'X-Shopify-Storefront-Access-Token': process.env.REACT_APP_SHOPIFY_STOREFRONT_ACCESS_TOKEN,
          Accept: 'application/graphql',
        },
        fetch,
      }) as unknown as ApolloLink,
    ]),
  })
}
