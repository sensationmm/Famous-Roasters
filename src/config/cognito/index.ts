import useAuth from './auth.hook'

export { useAuth as useAuth }
