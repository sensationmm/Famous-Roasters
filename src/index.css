@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body > div > div > main {
  min-height: 50vh;
}

:root {
  --swiper-theme-color: #646575 !important;
}

.video-container {
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 0;
  height: 0;
  overflow: hidden;
}

.video-container iframe,
.video-container object,
.video-container embed {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}