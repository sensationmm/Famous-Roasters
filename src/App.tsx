import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import {
  About,
  Account,
  Cart,
  Catalogue,
  Checkout,
  Contact,
  Error,
  Home,
  OurRoasters,
  Product,
  TasteFinder,
} from 'src/views'

import { i18n } from './config'

function App() {
  return (
    <I18nextProvider i18n={i18n}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/about" element={<About />} />
          <Route path="/account" element={<Account />} />
          <Route path="/cart" element={<Cart />} />
          <Route path="/catalogue" element={<Catalogue />} />
          <Route path="/checkout" element={<Checkout />} />
          <Route path="/contact" element={<Contact />} />
          <Route path="/our-roasters" element={<OurRoasters />} />
          <Route path="/product/:id" element={<Product />} />
          <Route path="/taste-finder" element={<TasteFinder />} />
          <Route path="*" element={<Error />} />
        </Routes>
      </BrowserRouter>
    </I18nextProvider>
  )
}

export default App
