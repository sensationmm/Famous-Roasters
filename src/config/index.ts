import { apolloClient } from './apollo'
import i18n from './i18n'

export { i18n as i18n }
export { apolloClient as apolloClient }
