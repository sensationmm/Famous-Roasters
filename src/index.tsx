import './index.css'

import React from 'react'
import { createRoot } from 'react-dom/client'

import App from './App'

const container = document.getElementById('root') as Element
const root = createRoot(container)
if (process.env.NODE_ENV === 'development') {
  root.render(
    <React.StrictMode>
      <App />
    </React.StrictMode>,
  )
} else {
  root.render(<App />)
}
