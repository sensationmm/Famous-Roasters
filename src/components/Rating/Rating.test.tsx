import { render, screen, waitFor } from '@testing-library/react'
import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { i18n } from 'src/config'

import { Rating } from '.'

describe('Rating component', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n} data-testId="">
        <Rating value={5} setValue={jest.fn} />
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('fires change event', async () => {
    const mockSetValue = jest.fn()
    render(
      <I18nextProvider i18n={i18n} data-testId="">
        <Rating value={5} setValue={mockSetValue} />
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    const button = await screen.findByTestId('rating-3')
    button.click()
    expect(mockSetValue).toHaveBeenCalledWith(3)
  })
})
