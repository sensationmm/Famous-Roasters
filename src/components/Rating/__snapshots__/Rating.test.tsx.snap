// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component Renders correctly 1`] = `
<div>
  <div>
    <div
      class="grid grid-cols-[repeat(5,_50px)] gap-6"
    >
      <div
        class="flex flex-col justify-center items-center"
        data-testid="rating-1"
      >
        <span
          class="font-normal text-coreUI-text-secondary mb-3 flex flex-col justify-center h-6 text-center text-xs leading-4 font-semibold"
        >
          Gar nicht gut
        </span>
        <button
          class="block w-11 h-11 rounded-full bg-coreUI-background-images font-bold cursor-pointer transition-all"
          id="rating-1"
        >
          1
        </button>
      </div>
      <div
        class="flex flex-col justify-center items-center"
        data-testid="rating-2"
      >
        <span
          class="font-normal text-coreUI-text-secondary mb-3 flex flex-col justify-center h-6 text-center text-xs leading-4 font-semibold"
        >
          Nicht gut
        </span>
        <button
          class="block w-11 h-11 rounded-full bg-coreUI-background-images font-bold cursor-pointer transition-all"
          id="rating-2"
        >
          2
        </button>
      </div>
      <div
        class="flex flex-col justify-center items-center"
        data-testid="rating-3"
      >
        <span
          class="font-normal text-coreUI-text-secondary mb-3 flex flex-col justify-center h-6 text-center text-xs leading-4 font-semibold"
        >
          Ok
        </span>
        <button
          class="block w-11 h-11 rounded-full bg-coreUI-background-images font-bold cursor-pointer transition-all"
          id="rating-3"
        >
          3
        </button>
      </div>
      <div
        class="flex flex-col justify-center items-center"
        data-testid="rating-4"
      >
        <span
          class="font-normal text-coreUI-text-secondary mb-3 flex flex-col justify-center h-6 text-center text-xs leading-4 font-semibold"
        >
          Gut
        </span>
        <button
          class="block w-11 h-11 rounded-full bg-coreUI-background-images font-bold cursor-pointer transition-all"
          id="rating-4"
        >
          4
        </button>
      </div>
      <div
        class="flex flex-col justify-center items-center"
        data-testid="rating-5"
      >
        <span
          class="font-normal text-coreUI-text-secondary mb-3 flex flex-col justify-center h-6 text-center text-xs leading-4 font-semibold"
        >
          Sehr gut
        </span>
        <button
          class="block w-11 h-11 rounded-full bg-primary font-bold cursor-pointer transition-all"
          id="rating-5"
        >
          5
        </button>
      </div>
    </div>
    <span
      class="block font-normal text-coreUI-text-secondary pt-4 text-xs leading-4"
    >
      Bewerte von 1 bis 5, wie gut du den Kaffee fandest.
    </span>
  </div>
</div>
`;
